keyboards/crkbd/keymaps/mcrown/oled.c

```diff
@@ -0,0 +1,354 @@
+            '#',         '$',      '%',      '^',      '&',      '*',     '(',      ')',     0x00,      0x00,    0x00,     0x00,     0x00,      '_',      '+',      '{',         /* 4 */
+        /*          |         |         |         |         |         |         |         |         |         |         |         |         |         |         |         |           */
+            '}',         '|',     0x00,     0x00,     0x00,      '~',    0x00,     0x00,     0x00,      0x00,    0x00,     0x00,     0x00,     0x00,     0x00,     0x00,         /* 5 */
+        /*          |         |         |         |         |         |         |         |         |         |         |         |         |         |         |         |           */
+           0x00,        0x00,     0x00,     0x00,     0x7F,     0x00,    0x00,     0x01,     0x02,      0x03,    0x04,     0x00,     0x00,     0x00,     0x00,     0x00,         /* 6 */
+        /*          |         |         |         |         |         |         |         |         |         |         |         |         |         |         |         |           */
+           0x00,        0x00,     0x00,     0x00,     0x00,     0x00,    0x00,     0x00,     0x00,      0x00,    0x00,     0x00,     0x00,     0x00,     0x05,     0x00,         /* 7 */
+};
+
+/** @brief maps the keycode to get the ascii value.
+ *
+ *  If any argument is invalid, the function has no effect.
+ *
+ *  @param keycode value of the pressed key.
+ *  @return ascii value of the pressed key or a special value for non-ascii keys.
+ */
+inline static char get_ascii(int16_t keycode){
+    uint8_t ascii_idx=0x00;
+
+    if(keycode<KC_F1){
+        ascii_idx=code_to_ascii[(uint8_t)keycode];
+    }else if(keycode<KC_KP_ENTER){
+        ascii_idx=code_to_ascii[SPECIAL_KEYS_SHIFT(keycode)];
+    }else if(KC_LANG1==keycode){
+        ascii_idx=code_to_ascii[ALT_CODE];
+    }else if( QK_LSFT==(QK_LSFT&keycode) ){
+        ascii_idx=code_to_ascii[RM_LSFT(keycode)];
+    }
+
+    return ascii_t[ascii_idx];
+}
+
+/** @brief detect retuns the rotation of the display based on the keyboard side.
+ *
+ *  If any argument is invalid, the function has no effect.
+ *
+ *  @param oled_rotation_t rotation
+ *  @return rotation of the display.
+ */
+oled_rotation_t oled_init_user(oled_rotation_t rotation){
+    oled_rotation_t oled_rot=OLED_ROTATION_180;
+
+    if(true==is_master){
+#ifdef OLED_VERTICAL
+        oled_rot=OLED_ROTATION_270;
+#else
+        oled_rot=OLED_ROTATION_0;
+#endif
+    }
+    return oled_rot;
+}
+
+/** @brief renders the keylog string and display it. This function also toggles the cursor.
+ *
+ *  If any argument is invalid, the function has no effect.
+ *
+ *  @param void.
+ *  @return void.
+ */
+static void render_keylogger_status(void){
+    static bool cursor_f=true;
+
+    if(timer_elapsed32(cursor_oled_timer) > 300){
+        cursor_oled_timer = timer_read32();
+        cursor_f=TOGGLE_BOOL_VAR(cursor_f);
```

Comment:
```suggestion
        cursor_f=!cursor_f;
```
I think it's easier this way than using a custom macro.

Reply:
Although that would work I'd rather to keep like this since booleans are no part o C standard and if this part of the code is reuse in a different compiler, that would lead to other issues. Here, we are just toggling a bit (first bit, depending how AVR compiler handles this boolean variables)